The infamous Vuforia mod!

Windwoes opened the pull request:
Need I say more? :D

Frogbots4634 committed:
Remove buildings & teapot bloat

Frogbots4634 committed:
Remove stones & chips bloat

Frogbots4634 committed:
Replace real Vuforia lib with a dummy one

Frogbots4634 committed:
Merge branch 'develop' of https://github.com/OpenFTC/OpenFTC-app-turbo into vuforiaMod

Merge splash screen

Frogbots4634 committed:
The mod itself

ItsHarper commented on FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java:
```diff
@@ -1,17 +1,175 @@
+    {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("libVuforia.so corrupted!");
+        builder.setMessage("libVuforia.so is present in the FIRST folder on the SDcard, however, the MD5 sum does not match what is expected.");
```
"checksum", not "sum"

ItsHarper commented on RobotCore/src/main/java/org/firstinspires/ftc/robotcore/internal/vuforia/VuforiaLocalizerImpl.java:
```diff
@@ -741,21 +744,22 @@ protected void updateActivityOrientation() {
             GLES20.glTexParameterf(GLES20.GL_TEXTURE_2D, GLES20.GL_TEXTURE_MAG_FILTER, GLES20.GL_LINEAR);
             GLES20.glTexImage2D(GLES20.GL_TEXTURE_2D, 0, GLES20.GL_RGBA, t.mWidth, t.mHeight, 0, GLES20.GL_RGBA, GLES20.GL_UNSIGNED_BYTE, t.mData);
-        }
+        }*/
```
~~Why is this stuff commented out? It doesn't look like it's related to the deleted assets to me.~~

Nevermind, I get it now that I've looked at the change in its proper context.

Windwoes replied:
Yeah, that's the stuff that was causing the crash problem earlier

ItsHarper commented on FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java:
```diff
@@ -1,17 +1,175 @@
+    {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("libVuforia.so not found!");
+        builder.setMessage("Please copy it to the FIRST folder on the root of the sdcard.");
```
The real libVuforia.so should be included in the repo (maybe the root of the doc folder?), and this message should tell the user exactly where to find it.

ItsHarper commented on FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java:
```diff
@@ -1,17 +1,175 @@
+    private boolean shouldProceedWithRcLaunch()
+    {
+        libInProtectedStorage = new File("/data/data/com.qualcomm.ftcrobotcontroller/extra/libVuforiaReal.so");
+        protectedExtraFolder = new File("/data/data/com.qualcomm.ftcrobotcontroller/extra/");
```
There's got to be an API to get the location of our app folder, rather than hard-coding it.

ItsHarper commented on FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java:
```diff
@@ -1,17 +1,175 @@
+    }
+
+    private boolean shouldProceedWithRcLaunch()
+    {
```
I don't love all the side effects that come from this method, _especially_ when it has such an innocuous name that implies that it's just doing checks. I also don't like the true/false error state. It only works because this method is doing too much (its core function, _and_ handling its errors, _and_ showing UI as a result).

I'd prefer to see it set up this way: this method should be called `setupFilesForLaunch()` (or similar), and if any errors happen, throw a custom checked exception. Let onCreate decide how to handle them.

ItsHarper reviewed (changes requested):
Wonderful work! I've got just a few changes to request for now.

Frogbots4634 committed:
Use API to get paths

Frogbots4634 committed:
Add libVuforia.so to 'doc' folder
Switch to throwing custom checked exceptions in Splash Activity
Add hyperlink to file on Github in "not found" dialog

FROGbots-4634 commented:
@NoahAndrews Ok I think I've taken care of all the changes you requested

noahAndrews committed:
tiny tweaks

ItsHarper commented on OpenFTC/src/main/java/org/openftc/exceptions/VuforiaNotFoundOnSdcardException.java:
```diff
@@ -0,0 +1,5 @@
+package org.openftc.exceptions;
+
+public class VuforiaNotFoundOnSdcardException extends Exception
```
this can just be VuforiaNotFoundException

ItsHarper commented on OpenFTC/src/main/java/org/openftc/exceptions/VuforiaCurruptedException.java:
```diff
@@ -0,0 +1,5 @@
+package org.openftc.exceptions;
+
+public class VuforiaCurruptedException extends Exception
```
"corrupted"

ItsHarper commented on FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java:
```diff
@@ -1,17 +1,195 @@
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
```
Clean up your unsused imports please :)

ItsHarper reviewed (changes requested):
Tiny tweaks that I tried to make myself real quick, but that got rejected by the server. Your fixes from last round were great.

FROGbots-4634 commented:
@NoahAndrews I think they got rejected by the server because I forgot to check the allow edits from maintainers box.

Try pushing again.

NoahAndrews commented:
Awesome, done. Now I just need to test it properly beyond the error screen. :P

noahAndrews committed:
Clarify vuforia error messages

noahAndrews committed:
Switch to permanent GitHub link
